sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/util/ActiveWindowSet.java

```diff
@@ -122,16 +124,23 @@ void onMerge(Collection<W> toBeMerged, Collection<W> activeToBeMerged, W mergeRe
   boolean isActive(W window);
 
   /**
-   * If {@code window} is not already known to be ACTIVE, MERGED or EPHEMERAL then add it
-   * as NEW. All NEW windows will be accounted for as ACTIVE, MERGED or EPHEMERAL by a call
-   * to {@link #merge}.
+   * Called when an incoming element indicates it is a member of {@code window}, but before we
+   * have started processing that element. If {@code window} is not already known to be ACTIVE,
+   * MERGED or EPHEMERAL then add it as NEW.
+   */
+  void seenWindow(W window);
+
+  /**
+   * Called when a window is known to be ACTIVE. Ensure that if it is NEW then it becomes ACTIVE
+   * (with itself as its only state address window).
    */
-  void addNew(W window);
+  void usingActiveWindow(W window);
```

Comment:
`usingActiveWindow` makes me think this should return something. Maybe `markWindowActive` or `markActiveWindowUsed`?